src/features/metadata-based-view/MetadataQueryAPIHelper.js

```diff
@@ -153,23 +156,26 @@ export default class MetadataQueryAPIHelper {
         const { metadata } = metadataEntry;
         return {
             ...metadataEntry,
-            metadata: this.flattenMetadata(metadata),
+            metadata: this.isV2 ? metadata : this.flattenMetadata(metadata),
         };
     };
 
     filterMetdataQueryResponse = (response: MetadataQueryResponseData): MetadataQueryResponseData => {
         const { entries = [], next_marker } = response;
         return {
-            entries: entries.filter(entry => getProp(entry, 'type') === ITEM_TYPE_FILE), // return only file items
+            entries: this.isV2 ? entries : entries.filter(entry => getProp(entry, 'type') === ITEM_TYPE_FILE), // return only file items
             next_marker,
         };
     };
 
     getFlattenedDataWithTypes = (templateSchemaResponse?: MetadataTemplateSchemaResponse): Collection => {
         this.metadataTemplate = getProp(templateSchemaResponse, 'data');
+
         const { entries, next_marker }: MetadataQueryResponseData = this.metadataQueryResponseData;
+
         return {
             items: entries.map<BoxItem>(this.flattenResponseEntry),
+            metadataTemplate: this.metadataTemplate,
```

Comment:
feels weird to include this property in the return of `getFlattenedDataWithTypes`